src/connectors/toggle/connectToggle.js

```diff
@@ -275,6 +275,52 @@ export default function connectToggle(renderFn, unmountFn) {
 
         return nextState;
       },
+
+      getWidgetState(fullState, { state }) {
+        const isRefined = state.isDisjunctiveFacetRefined(attributeName, on);
+
+        if (
+          !isRefined ||
+          (fullState &&
+            fullState.toggle &&
+            fullState.toggle[attributeName] === isRefined)
+        ) {
+          return fullState;
+        }
+
+        return {
+          ...fullState,
+          toggle: {
+            ...fullState.toggle,
+            [attributeName]: isRefined,
+          },
+        };
+      },
+
+      getWidgetSearchParameters(searchParam, { uiState }) {
+        const isRefined = Boolean(
+          uiState.toggle && uiState.toggle[attributeName]
+        );
+
+        if (isRefined) {
+          if (hasAnOffValue)
+            return searchParam
+              .removeDisjunctiveFacetRefinement(attributeName, off)
+              .addDisjunctiveFacetRefinement(attributeName, on);
+
+          return searchParam.addDisjunctiveFacetRefinement(attributeName, on);
+        } else {
```

Comment:
The else branch is useless, we early return in the condition.

Reply:
Old habits :(